### ISSUE TITLE
'transform' method bug when the node is scaled

### ISSUE THREAD

**Issue by rogeriodec** [godotengine/godot#26938]
**Godot version:**
3.1 rc2

**OS/device including version:**
Windows 10 x64

**Issue description:**
I have this simple Cube MeshInstance (no scale):

![](https://cdn.discordapp.com/attachments/212250894228652034/554789135047458840/unknown.png)

I want to duplicate this cube and leave it exactly next to the original cube.
For this I use this code:
```
extends Spatial

func _ready() -> void:
    var m = $MeshInstance.duplicate()
    var mesh_width = $MeshInstance.transform.basis.x.x * 2 # width
    m.translate(Vector3(mesh_width, 0, 0))
    add_child(m)

```
The second cube is placed in the right way:
![](https://cdn.discordapp.com/attachments/212250894228652034/554789224012972042/unknown.png)

So I changed the 'x' scale from 1 to 0.3.
![](https://cdn.discordapp.com/attachments/212250894228652034/554789315738075322/unknown.png)

And when I run the same code, look:
![](https://cdn.discordapp.com/attachments/212250894228652034/554789385564848129/unknown.png)

What's the problem with the translate method? 
Why it doesn't work when a mesh is scaled?
How to solve this?

**Minimal reproduction project:**
[translate-bug.zip](https://github.com/godotengine/godot/files/2954457/translate-bug.zip)

**Comment by clayjohn** [comment]
"translate" translates the object in local space. So when you scale the mesh, the translation is scaled by that amount. If you want to set translation directly do "m.translation.x += mesh_width"

**Comment by akien-mga** [comment]
Fixed by #30049.
